dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java

```diff
@@ -220,6 +240,51 @@ private List<FileInfo> getFilePreviewContent(Context context, Bitstream bitstrea
         return fileInfos;
     }
 
+    /**
+     * Create sub map for preview content and file info.
+     * @param sourceMap  parent sub map
+     * @param creator    creator function
+     * @return    created sub map
+     */
+    private <T, U> Hashtable<String, T> createSubMap(Map<String, U> sourceMap, Function<U, T> creator) {
+        Hashtable<String, T> sub = new Hashtable<>();
+        if (sourceMap != null) {
```

Comment:
So when the sourceMap is null it will return an empty Map right? I suggest to refactor this to 
```
if (sourceMap == null) {
return sub;
}
```